code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -121,7 +125,13 @@
 
 /mob/living/silicon/robot/proc/is_component_functioning(module_name)
 	var/datum/robot_component/C = components[module_name]
-	return C && C.installed == 1 && C.toggled && C.is_powered()
+	return C && C.installed == 1 && C.toggled && C.is_powered() && C.not_disabled()
```

Comment:
You could probably just do `return C && C.installed == 1 && C.toggled && C.is_powered() && !C.component_disabled`